cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -490,26 +490,14 @@ public String parseLogging(Logging logging, ConfigMap externalCm) {
 
             return createLog4jProperties(newSettings);
         } else if (logging instanceof ExternalLogging) {
-            if (((ExternalLogging) logging).getValueFrom() == null) {
-                if (externalCm != null && externalCm.getData() != null && externalCm.getData().containsKey(getAncillaryConfigMapKeyLogConfig())) {
-                    return maybeAddMonitorIntervalToExternalLogging(externalCm.getData().get(getAncillaryConfigMapKeyLogConfig()));
-                } else {
-                    log.warn("ConfigMap {} with external logging configuration does not exist or doesn't contain the configuration under the {} key. Default logging settings are used.",
-                            ((ExternalLogging) getLogging()).getName(),
-                            getAncillaryConfigMapKeyLogConfig());
-                    return createLog4jProperties(getDefaultLogConfig());
-                }
+            if (externalCm != null && externalCm.getData() != null && externalCm.getData().containsKey(((ExternalLogging) logging).getValueFrom().getConfigMapKeyRef().getKey())) {
```

Comment:
Do we need to check this `((ExternalLogging) logging).getValueFrom().getConfigMapKeyRef().getKey()` for some null values?

Reply:
```            if (externalCm != null && externalCm.getData() != null && '((ExternalLogging) logging).getValueFrom().getConfigMapKeyRef().getKey() != null' && externalCm.getData().containsKey(((ExternalLogging) logging).getValueFrom().getConfigMapKeyRef().getKey())) {``` I will add this check here

Reply:
Like we don't need any other else if case i guess

Reply:
@scholzj I have applied the suggestion